fawltydeps/settings.py

```diff
@@ -93,6 +93,27 @@ def parse_path_or_stdin(arg: str) -> PathOrSpecial:
     return Path(arg)
 
 
+default_ignore_unused = set(
+    [  # Development tools not meant to be imported
+        "black",
+        "codespell",
+        "colorama",
```

Comment:
As I think was already suggested somewhere else: `colorama` does not belong in this list, as it's not a stand-alone tool (rather a transitive library dependency of many other tools).

Reply:
`colorama` is a direct dependency in our pyproject.toml,
```toml
[tool.poetry.group.format.dependencies]
colorama = "^0.4.6"
...
[tool.poetry.group.dev.dependencies]
colorama = "^0.4.6"
```
In order to make our CI (specially `self_test`) pass, I added it to the list.

Also, this brings me to think, as our list of development dependencies continues to grow, should we consider implementing functionality that allows users to add other tools to the `DEFAULT_IGNORE_UNUSED` list without overwriting it entirely? This could provide more flexibility for users while maintaining our core list of dependencies.
Any ideas? @jherland @Nour-Mws @mknorps

Reply:
> colorama is a direct dependency in our pyproject.toml

@jherland it seems it has been added in this commit https://github.com/tweag/FawltyDeps/commit/b86a20eb7956691f6b0851fb494cab39835f4c1a.

> Also, this brings me to think, as our list of development dependencies continues to grow, should we consider implementing functionality that allows users to add other tools to the DEFAULT_IGNORE_UNUSED list without overwriting it entirely? This could provide more flexibility for users while maintaining our core list of dependencies.
Any ideas?

I rather prefer the user takes full responsibility of declaring this list the moment they decide to add a dependency to it. I believe having an in-between state is in the realm of diminishing returns of efforts to functionality.

Reply:
> > colorama is a direct dependency in our pyproject.toml
> 
> @jherland it seems it has been added in this commit [b86a20e](https://github.com/tweag/FawltyDeps/commit/b86a20eb7956691f6b0851fb494cab39835f4c1a).

Ah, I see, `colorama` is an _optional_ dependency of `isort` and `black`: If it is available, it is automatically used to colorize their output, but they otherwise do not depend on it.

This makes it an oddball when considering `DEFAULT_IGNORE_UNUSED`: It is _not_ a tool that is meant to be invoked from the command-line, but there are still arguments for why you'd want to depend on it without (yourself) `import`ing it...

Actually, when I reconsider this now, I think we could have done better in b86a20e: Instead of adding `colorama` as a direct dependency to FawltyDeps, we should make use of the fact that both `black` and `isort` provide optional/extra dependency groups ([here](https://github.com/psf/black/blob/e9356c1ff0083aea4416bf1d3e29748634bb4f7f/pyproject.toml#L76) and [here](https://github.com/PyCQA/isort/blob/e35a55f6ab8473003d68a8b2ffebdd08fbb2b61b/pyproject.toml#L50), respectively) that bring in `colorama` as a transitive dependency.

I think this - combined - with dropping `colorama` from `DEFAULT_IGNORE_UNUSED` is the cleanest solution.

In other words, try something like this diff on our `pyproject.toml`:
```diff
diff --git a/pyproject.toml b/pyproject.toml
index 48d6cd6..bf1450c 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -74,12 +74,11 @@ types-setuptools = "^65.6.0.2"
 optional = true
 
 [tool.poetry.group.format.dependencies]
-black = "^22"
-colorama = "^0.4.6"
+black = {extras = ["colorama"], version = "^22"}
 isort = [
     # isort 5.12.0 drops support for Python v3.7:
-    {version = "^5.10", python = ">=3.8"},
-    {version = ">=5.10,<5.12.0", python = "<3.8"},
+    {extras = ["colors"], version = "^5.10", python = ">=3.8"},
+    {extras = ["colors"], version = ">=5.10,<5.12.0", python = "<3.8"},
 ]
 codespell = "^2.2.4"
 
@@ -91,9 +90,8 @@ optional = true
 # something to the above groups (i.e. something that targets a specific purpose
 # (e.g. a CI action), consider whether it's also useful to have this available
 # in a developers environment
-black = "^22"
+black = {extras = ["colorama"], version = "^22"}
 codespell = "^2.2.4"
-colorama = "^0.4.6"
 hypothesis = "^6.68.2"
 mypy = "^1.0.1"
 nox = "^2022.11.21"
```

> > Also, this brings me to think, as our list of development dependencies continues to grow, should we consider implementing functionality that allows users to add other tools to the DEFAULT_IGNORE_UNUSED list without overwriting it entirely? This could provide more flexibility for users while maintaining our core list of dependencies.
> > Any ideas?
> 
> I rather prefer the user takes full responsibility of declaring this list the moment they decide to add a dependency to it. I believe having an in-between state is in the realm of diminishing returns of efforts to functionality.

TL;DR: Yes, I think I agree with @Nour-Mws here. This has been discussed somewhat before in https://github.com/tweag/FawltyDeps/issues/328, and I suggest that we continue this part of the discussion there...

Reply:
Thanks @jherland for digging into this! I agree that making `colorama` a transitive dependency is a neat solution.
It is addressed in https://github.com/tweag/FawltyDeps/pull/365/commits/c263fbdeba33e49d1f5b2c38a82825d4de60d019. 
For the second point, it makes great sense that we wait to see if there is some demand for extending/combining some directives instead always/only replacing. :100:

Reply:
Your changes look good to me.

The fact that the discussion on extend/combine vs. replace has come up (at least) twice  in different contexts indicates that there is _some_ demand, but let's track/prioritize that separately in #328. 👍🏻